### DISCUSSION TITLE
CPE matching without vendor specified

### DISCUSSION THREAD

**Discussion by Trakeur** [DependencyTrack/dependency-track#2823] [Q&A]
Hi everyone,
I would have liked to know if the non-finding of CVEs associated with a CPE whose vendor field is missing, in my case with : 
cpe:2.3:a:*:perl:5.36.0:*:*:*:*:*:*:*
, is normal (in theory 6 vulnerabilities detected) ? 

Even with fuzzy CPE matching enabled.

Is there a way for DT to detect vulnerabilities associated with the component's name and version but not the vendor? (even if it can bring false positives).

**Comment by stevespringett** [comment]
`cpe:2.3:a:perl:5.36.0:::::::` is an invalid CPE. The correct CPE should be `cpe:2.3:a:perl:perl:5.6.0:*:*:*:*:*:*:*`. 

If this is being stored in DT, it might be the reason why vulns are not being identified. And yes, I believe DT should be able to identify vulns via fuzzy matching without a vendor/groupId

**Reply by Trakeur** [reply]
It is indeed an invalid cpe, it's just that the `*`(star) were removed when I wrote the message. I used exactly the CPE mentioned just below by @officerNordberg, but no result. if I specify the vendor there the vulnerabilities are detected.

**Comment by officerNordberg** [comment]
> The correct CPE should be `cpe:2.3:a:perl:perl:5.6.0:*:*:*:*:*:*:*`

or `cpe:2.3:a:*:perl:5.6.0:*:*:*:*:*:*:*` should work as a vendor/groupid free fuzzy matched option.  I worked specifically to make that possible but ymmv.  I just tested my suggested cpe and it matched on 21 vulns.

**Reply by Trakeur** [reply]
I used the same cpe as you but no vulnerabilities detected. Which fuzzing option did you activate?

**Reply by officerNordberg** [reply]
Just the first "Enable fuzzy CPE matching. Helps with inconsistent NVD data, highlighting missing risks but also increasing false positives"

**Reply by officerNordberg** [reply]
You may need to rebuild your Vulnerable Software Index.  Fuzzy Matching relies entirely on Lucene indices being up-to-date and operational.

**Reply by Trakeur** [reply]
So I restarted dependency-track to take the changes into account (several hours ago now), and indeed new vulnerabilities have been detected, but still not on the component given as an example of Perl, even though there are vulnerabilities on it.

I'll try to reupload my SBOM, maybe it will work.

**Comment by officerNordberg** [comment]
What version are you on? I tried it again. `cpe:2.3:a:*:perl:5.6:*:*:*:*:*:*:*` has some false-positives but mostly matches Perl as expected. Also as expected`cpe:2.3:a:perl:perl:5.6:*:*:*:*:*:*:*` improves the results.

**Comment by officerNordberg** [comment]
As far as I can tell 5.36.0 has no known CVEs. These are all the `running with perl:perl:*` CPEs and not a vulnerability in the language/runtime/version [NVD Search Link](https://nvd.nist.gov/vuln/search/results?form_type=Advanced&results_type=overview&isCpeNameSearch=true&seach_type=all&query=cpe:2.3:a:perl:perl:5.36.0:*:*:*:*:*:*:*)
